oauth2-server/src/main/java/com/clouway/oauth2/ClientController.java

```diff
@@ -25,14 +25,14 @@ public Response handleAsOf(Request request, ClientCredentials credentials, DateT
 
     // Client was not authorized
     if (!possibleResponse.isPresent()) {
-      return OAuthError.unauthorizedClient();
+      return OAuthError.unauthorizedClient("A client with id:" + credentials.clientId() + " was either not found or is not authorized.");
     }
 
     Client client = possibleResponse.get();
 
     // Client credentials did not match?
     if (!client.credentialsMatch(credentials)) {
-      return OAuthError.unauthorizedClient();
+      return OAuthError.unauthorizedClient("Client credentials did not match.");
```

Comment:
The error should be same for unknown and non matching secret as it will not provide information to the caller which to be used for brute force: 

OAuthError.unauthorizedClient(String.format("Unknown client '%s'", clientId))

Reply:
Agreed changed.